backend/src/components/institute/institute.js

```diff
@@ -241,6 +241,103 @@ async function getCachedSchoolBySchoolID(req, res) {
   }
 }
 
+async function addSchool(req, res) {
+  try {
+    const token = getBackendToken(req);
+    let hasSchoolAdminRoleAuth;
+
+    if(req.body.schoolCategoryCode === 'INDEPEND' || req.body.schoolCategoryCode === 'INDP_FNS'){
+      hasSchoolAdminRoleAuth = req.session.roles.includes('SCHOOL_ADMIN') || req.session.roles.includes('SCHOOL_INDEPENDENT_ADMIN');
+    } else {
+      hasSchoolAdminRoleAuth = req.session.roles.includes('SCHOOL_ADMIN');
+    }
+
+    if (hasSchoolAdminRoleAuth) {
+      return res.status(HttpStatus.UNAUTHORIZED).json({
+        message: 'You do not have the required access for this function'
+      });
+    }
```

Comment:
I think the condition would be !hasSchoolAdminRoleAuth
This check can also be combined with hasSchoolAdminRole check